src/lib/OS/SimOS.cc

```diff
@@ -244,17 +254,8 @@ uint64_t SimOS::createProcess(span<char> instructionBytes) {
         static_cast<uint64_t>(0b10100), 8};
   }
 
-  // If this is the initial process (tid = 0) then add to the scheduledProcs_
-  // queue as only processes in scheduledProcs_ can be scheduled onto an idle
-  // core (all cores begin in an idle state).
-  // Otherwise, add the new process to the waitingProcs_ queue.
-  if (tid == 0) {
-    processes_[tid]->status_ = procStatus::scheduled;
-    scheduledProcs_.push(processes_[tid]);
-  } else {
-    processes_[tid]->status_ = procStatus::waiting;
-    waitingProcs_.push(processes_[tid]);
-  }
+  processes_[tid]->status_ = procStatus::waiting;
+  waitingProcs_.push(processes_.find(tid)->second);
```

Comment:
why has this been changed from `waitingProcs_.push(processes_[tid]);`?

Reply:
This was re-written so by mistake. Will change back